Improved Graph overview and showcase actual data. Altered structure for mutation indication.

joswinter opened the pull request:
#54 Improve Graph overview: Scaling of blocks etc.

I will create a pull request for Mutation Indication later this week because I can now test on actual data.

joswinter committed:
Slight refactor for clarity.

joswinter committed:
Fix javadoc.

joswinter committed:
Create interface for segment content. Update Tileview, VertexView.

joswinter committed:
Implement dividing edges.

joswinter committed:
Restore ordering in sequencesegment.

joswinter committed:
Fix javadoc/formatting.

joswinter committed:
Small styling changes.

joswinter committed:
Remove redundant variable.

joswinter committed:
Add tests for DefaultParser, Sequence, SequenceSegment, JUnitTGraphAdapter.

joswinter committed:
Improve sequenceSegment compareTo.

joswinter committed:
Tiny modification to shut checkstyle up.

joswinter committed:
Fixed javadoc mutation

joswinter committed:
add @Deprecated to showcase functions

joswinter committed:
Fixed showcase graph file directory

joswinter committed:
Removed unused error prone function TileController

joswinter committed:
Renamed getSource(s) and getSink(s)

joswinter committed:
Renamed divideEdge -> splitEdge

joswinter committed:
Removed unclear newline

joswinter committed:
Fixed Unused imports

joswinter committed:
Fixed javadoc PhylogeneticTreeItem

jorenham commented on src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java:
```diff
@@ -69,6 +74,7 @@ public final boolean addEdge(final V source, final V destination) {
         try {
```
Using a `try .. catch` block instead of pre-checking with `contains` is bad code style and also not very _fail fast_.

joswinter committed:
Remove try catch, use contains JGraphTGraphAdapter

jorenham commented on src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java:
```diff
@@ -36,180 +35,164 @@
+                    && segmentFree(index, segment)) {
+                segmentInsert(index, segment);
+                String text = segment.getContent().toString();
+                drawVertex(text, segment.getStart(), index, segment
```
Bad readability; you should create intermediate variables for the big arguments.

jorenham commented on src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java:
```diff
@@ -36,180 +35,164 @@
-
-            if ((i + w > lanes.size())
-                    && (lanes.get(i + w) <= segment.getStart())) {
+    private Boolean segmentFree(final int ind, final SequenceSegment segment) {
```
Minor inconsistency: Return type is `Boolean` and not primitive `boolean`

jorenham commented on src/main/java/nl/tudelft/lifetiles/graph/view/VertexView.java:
```diff
@@ -42,9 +37,19 @@
+     * Horizontal scale for each coordinate.
      */
-    private final double spacingX = 3;
+    private static final double HORIZONTALSCALE = 15;
```
I think these constants should be variables, with semantic zooming in mind

joswinter replied:
I think so too, but we shouldn't store the spacing and the scale of the vertices in all the vertices. It would be better if we would centralize the variable in another class, the TileView for example.

jorenham commented on src/main/java/nl/tudelft/lifetiles/graph/view/VertexView.java:
```diff
@@ -156,12 +122,12 @@ public final double getWidth() {
-        final double h = rectangle.getHeight();
-        clip.setWidth(w);
-        clip.setHeight(h);
+        final double width = rectangle.getWidth();
```
Minor inconsistency: `final` local variables. Either make all the appropriate local variables final or none of them.

jorenham commented on src/test/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegmentTest.java:
```diff
@@ -0,0 +1,54 @@
+        v1.setMutation(insertion);
+        assertEquals(insertion, v1.getMutation());
+    }
+}
```
Missing trailing newline.

joswinter committed:
Added intermediate variables  in TileView.drawVertexLane

Bad readability; you should create intermediate variables for the big
arguments.

joswinter committed:
TileView.segmentFree changed correct return type

Minor inconsistency: Return type is Boolean and not primitive boolean

joswinter committed:
Removed final keyword local variable VertexView

Minor inconsistency: final local variables. Either make all the
appropriate local variables final or none of them.

joswinter committed:
Added Missing trailing newline SequenceSegmTest

Missing trailing newline.

joswinter committed:
Removed Line Trailing Spaces, TileView

[ERROR] src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java[95]
(regexp) RegexpSingleline: Line has trailing spaces.
